model/account/accounts.go

```diff
@@ -118,6 +119,50 @@ func GetTriggers(jobsSystem job.JobSystem, db prefixer.Prefixer, accountID strin
 	return toDelete, nil
 }
 
+// CleanEntry is a struct with an account and its associated trigger.
+type CleanEntry struct {
+	Account          *Account
+	Trigger          job.Trigger
+	ManifestOnDelete bool // the manifest of the konnector has a field "on_delete_account"
+	Slug             string
+}
+
+// CleanAndWait deletes the accounts. If an account is for a konnector with
+// "on_delete_account", a job is pushed and it waits for the job success to
+// continue. Finally, the associated trigger can be deleted.
+func CleanAndWait(inst *instance.Instance, toClean []CleanEntry) error {
+	jobsSystem := job.System()
+	for _, entry := range toClean {
+		acc := entry.Account
+		acc.ManualCleaning = true
+		oldRev := acc.Rev() // The deletion job needs the rev just before the deletion
+		if err := couchdb.DeleteDoc(inst, acc); err != nil {
```

Comment:
Can't we delete the document last? We wouldn't need to keep its old revision and we would still have a reference to this account in case something goes awry.

Reply:
Well, we could do that here. But the `on_delete.js` script called to clean the account on the remote system is also called when an account is deleted by a user. So, if we do that, we push some complexity on this script, as it would have to manage two ways to be called. And I think it would add more complexity on the JS side that it would remove on the go side.